src/components/details-modal/DetailsModal.js

```diff
@@ -1,96 +1,108 @@
-import { Heading, Text, Flex, Image, Box } from '@chakra-ui/react';
-import { useParams } from 'react-router-dom';
+import { Heading, Text, Flex, Image, Box, Button } from '@chakra-ui/react';
 import Data from '../guest-homepage/Data';
 
-const DetailsModal = () => {
-    const { id } = useParams();
-    let idFixed = id.slice(1, id.length);
+const DetailsModal = ({ modal, itemID, setModal }) => {
+    if (!modal) {
+        return <></>;
+    } else {
+        const singleItem = Data.find((item) => item.id === +itemID);
```

Comment:
No need to do the finding of the item in the modal, we can pass the whole item object to this component, more info in the commends below